docs/querying/using-caching.md

```diff
@@ -0,0 +1,92 @@
+To use caching, you must first enable the settings for a service in the runtime properties. Afterward, you can control which queries use the cache or populate the cache on a per-query basis within the query context.
+
+The query context settings do not override the server settings. For example, it is possible to set a query context to use the cache, but if it is not enabled at the service level, there's no cache to use.
+
+## Enabling query caching on Brokers
+Brokers support both segment-level and whole-query result level caching.
+
+To control **segment caching** on the Broker, set the `useCache` and `populateCache`runtime properties. For example, to set the Broker to use and populate the segment cache for queries:
+```
+druid.broker.cache.useCache=true
+druid.broker.cache.populateCache=true
+```
+To control **whole-query caching** on the Broker, set the `useResultLevelCache` and `populateResultLevelCache` runtime properties. For example, to set the Broker to use and populate the whole-query cache for queries:
+```
+druid.broker.cache.useResultLevelCache=true
+druid.broker.cache.populateResultLevelCache=true
+```
+See [Broker caching](../configuration/index.md#broker-caching) for a description of all available Broker cache configurations.
+ 
+## Enabling query caching on Historicals
+Historicals only support **segment-level** caching. To control caching on the Historical, set the `useCache` and `populateCache` runtime properties. For example, to set the Historical to both use and populate the segment cache for queries:
+ ```
+ druid.broker.cache.useCache=true
+ druid.broker.cache.populateCache=true
+ ```
+See [Historical caching](../configuration/index.md#historical-caching) for a description of all available Historical cache configurations.
+ 
+## Enabling query caching on task executor services
+Task executor services, the Peon or the Indexer, only support **segment-level** caching. To control caching on a task executor service, set the `useCache` and `populateCache` runtime properties. For example, to set the Peon to both use and populate the segment cache for queries:
+
+```
+druid.realtime.cache.useCache=true
+druid.realtime.cache.populateCache=true
+```
+
+See [Peon caching](configuration/index.md#peon-caching) for a description of all available task executor service caching options.
+
+## Enabling caching in the query context
+After you enable caching for a service, set cache options for individual queries in the query [context](./query-context.md). For example, you can `POST` a Druid SQL request to the HTTP POST API and include the context as a JSON object:
+
+```
+{
+  "query" : "SELECT COUNT(*) FROM data_source WHERE foo = 'bar' AND __time > TIMESTAMP '2020-01-01 00:00:00'",
+  "context" : {
+    "useCache" : "true",
+    "populateCache" : "false"
+  }
+}
+```
+In this example `populateCache` is `false` because the query references data that is over a year old. For more information, see [Druid SQL client APIs](./sql.md#client-apis).
```

Comment:
`because the query references data that is over a year old. `

This reads as if the system is smart enough to make this call. I think we want to say something along the lines of the user making the query does not want to fill up the cache with results for old segments, so they may decide to disable populating the cache for old data, but still want to use the cache for recent data. In this scenario, they'd set the query context like this. What do you think?

Reply:
Great edit